gui/gui.go

```diff
@@ -84,6 +84,30 @@ func (ui *GUI) route() {
 	ui.router.HandleFunc("/admin", ui.PostAdmin).Methods("POST")
 	ui.router.HandleFunc("/backup", ui.PostBackup).Methods("POST")
 	ui.router.HandleFunc("/logout", ui.PostLogout).Methods("POST")
+
+	// Websocket endpoint allows the GUI to receive updated values
+	ui.router.HandleFunc("/ws", ui.RegisterWebSocket).Methods("GET")
+
+	socketUpdate := func() {
+		poolStats, err := ui.hub.FetchStats()
+		if err != nil {
+			log.Error(err)
+			return
+		}
+
+		ui.SendUpdatedValues(poolStats)
+	}
+
+	// Use a ticker to push updates through the socket periodically
+	ticker := time.NewTicker(RefreshRate)
```

Comment:
let's have the goroutine exit cleanly via a context. Also lets limit the scope of the ticker to the goroutine it's used in:

```
go func(ctx context){
	ticker := time.NewTicker(RefreshRate)
	defer ticker.Stop()

	for {
		select{
		case <-ticker.C:
			socketUpdate()

		case ctx.Done():
			return
		}
	}
}()
```